dummy-project/src/features.py

```
@@ -24,3 +24,18 @@ def normalize_column(df: pd.DataFrame, column: str) -> pd.DataFrame:
+def extract_keywords(df, column):
+    """Extract keywords from text."""
+    keywords = []
+    for text in df[column]:  # ⚠️ slow for large data
+        tokens = text.split()
+        keywords.append([t for t in tokens if len(t) > 5])
+    df['keywords'] = keywords
+    return df
+
+def one_hot_encode(df, column):
+    """One hot encode the specified column."""
+    dummies = pd.get_dummies(df[column])
```

Ruff (0.8.2)

Undefined name `pd`

(F821)